static/app/utils/url/validateLinkExists.ts

```diff
@@ -0,0 +1,23 @@
+import {addErrorMessage} from 'sentry/actionCreators/indicator';
+import {t} from 'sentry/locale';
+import {handleXhrErrorResponse} from 'sentry/utils/handleXhrErrorResponse';
+
+/**
+ * Checks if a URL exists by sending a HEAD request.
+ * @param url - The URL to check.
+ * @returns A promise that resolves to `true` if the URL exists, or `false` otherwise.
+ */
+export async function validateLinkExists(url: string): Promise<boolean> {
+  try {
+    const response = await fetch(url, {method: 'HEAD'}); // Using HEAD method to just check if the resource exists
+    if (response.ok) {
+      return true;
+    }
+    return false;
+  } catch (error) {
+    const message = t('Unable to validate ig the link exists');
+    handleXhrErrorResponse(message, error);
+    addErrorMessage(message);
```

Comment:
We should remove `addErrorMessage` from here so that it can be called only in cases where it's needed.

You mentioned that the first use-case for this util is inside a `handleSubmit` callback, but the hook could also be useful in a situation where, for example, we validate that an off-site link exists before the user clicks into it. In a case like that we wouldn't want to have addErrorMessage called because there's no user interaction yet.